core/jobs/batch_jobs/suggestion_stats_computation_jobs.py

```diff
@@ -268,6 +559,146 @@ def _generate_translation_contribution_model(
             translation_contributions_stats_model.update_timestamps()
             return translation_contributions_stats_model
 
+    @staticmethod
+    def _generate_translation_review_model(
+        entity_id: str,
+        translation: suggestion_registry.TranslationReviewStats
+    ) -> suggestion_models.TranslationReviewStatsModel:
+        """Generate translation review stats model from the domain object.
+
+        Args:
+            entity_id: str. The ID of the model.
+            translation: TranslationReviewStats. Domain object.
+
+        Returns:
+            TranslationReviewStatsModel. The created model.
+        """
+        language_code, contributor_user_id, topic_id = entity_id.split('.')
+        with datastore_services.get_ndb_context():
```

Comment:
Is this only needed to call `update_timestamps()`? If so, can we move the model creation out of the `with` block?

Reply:
Is that error thrown even if you only move the `suggestion_models.TranslationReviewStatsModel()` call? Like:

```
translation_review_stats_model = suggestion_models.TranslationReviewStatsModel(...)
with
  translation_review_stats_model.update_timestamps()
return translation_review_stats_model
```

Just trying to minimize the amount of code in the with block to keep it focused.

Reply:
Yes I am getting the same error when it is changed like below.

![image](https://user-images.githubusercontent.com/41409951/229574671-03e7a03f-cf89-41d9-bf6d-b8f41386949f.png)